src/store/models/taro.ts

```diff
@@ -54,6 +71,61 @@ const taroModel: TaroModel = {
     const balances = await api.listBalances(node);
     actions.setBalances({ node, balances });
   }),
+  getAllInfo: thunk(async (actions, node) => {
+    await actions.getAssets(node);
+    await actions.getBalances(node);
+  }),
+
+  mintAsset: thunk(
+    async (actions, payload, { injections, getStoreState, getStoreActions }) => {
+      const {
+        node,
+        assetType,
+        name,
+        amount,
+        metaData,
+        enableEmission,
+        skipBatch,
+        autoFund,
+      } = payload;
+
+      const network = getStoreState().network.networkById(node.networkId);
+      const lndNode = network.nodes.lightning.find(
+        n => n.name === node.lndName,
+      ) as LightningNode;
+      //fund lnd node
+      if (autoFund) {
+        await getStoreActions().lightning.depositFunds({
+          node: lndNode,
+          sats: TARO_MIN_LND_BALANCE.toString(),
+        });
+      }
+
+      //mint taro asset
+      const taroapi = injections.taroFactory.getService(node);
+      const req: TARO.MintAssetRequest = {
+        assetType,
+        name,
+        amount: assetType === PTARO.TARO_ASSET_TYPE.COLLECTIBLE ? 1 : amount,
+        metaData: Buffer.from(metaData, 'hex'),
+        enableEmission,
+        skipBatch,
+      };
+      const res = await taroapi.mintAsset(node, req);
+      (async () => {
```

Comment:
Why do you need this nested `async` function?

Reply:
This allows for the modal to close and to move on with business doing otherstuff. The mining and updating taro node balances takes about a second and I didn't see a reason why it couldn't be pushed to the background.
Maybe I'm impatient?

I was thinking progress bars might be nice as these operations are going on to give some feedback.

If you disagree I'm happy to remove the async.

Reply:
You can simply just remove the `await` keyword when calling `getStoreActions().bitcoind.mine()` which will have the same effect. The mining will occur in the background while the function execution will continue on without waiting. We do something similiar in `network.monitorStartup()`. Just please add a comment to indicate that the missing `await` is intentional and not a bug.

Reply:
Done.